care/facility/tasks/plausible_stats.py

```diff
@@ -0,0 +1,131 @@
+
+    response = request(
+        "GET",
+        url,
+        headers={
+            "Authorization": "Bearer " + settings.PLAUSIBLE_AUTH_TOKEN,
+        },
+    )
+
+    return response.json()
+
+
+def get_goal_event_stats(plausible_host, site_id, date, goal_name, event_name):
+    goal_filter = f"event:name=={goal_name}"
+
+    # pageview is a reserved goal in plausible which uses event:page
+    if goal_name == "pageview" and event_name == "page":
+        goal_event = "event:page"
+    else:
+        goal_event = f"event:props:{event_name}"
+
+    url = f"https://{plausible_host}/api/v1/stats/breakdown?site_id={site_id}&property={goal_event}&filters={goal_filter}&period=day&date={date}&metrics=visitors,events"
+
+    response = request(
+        "GET",
+        url,
+        headers={
+            "Authorization": "Bearer " + settings.PLAUSIBLE_AUTH_TOKEN,
+        },
+    )
+
+    return response.json()
+
+
+@shared_task
+def capture_goals():
+    if (
+        not settings.PLAUSIBLE_HOST
+        or not settings.PLAUSIBLE_SITE_ID
+        or not settings.PLAUSIBLE_AUTH_TOKEN
+    ):
+        logger.info("Plausible is not configured, skipping")
+        return
+    today = now().date()
+    yesterday = today - timedelta(days=1)
+    logger.info(f"Capturing Goals for {yesterday}")
+
+    for goal in Goals:
+        try:
+            goal_name = goal.formatted_name
+            goal_data = get_goal_stats(
+                settings.PLAUSIBLE_HOST,
+                settings.PLAUSIBLE_SITE_ID,
+                yesterday,
+                goal_name,
+            )
+            goal_object, _ = Goal.objects.get_or_create(
+                name=goal_name,
+            )
+            goal_entry_object, _ = GoalEntry.objects.get_or_create(
+                goal=goal_object,
+                date=yesterday,
+            )
+            goal_entry_object.visitors = goal_data["results"]["visitors"]["value"]
+            goal_entry_object.events = goal_data["results"]["events"]["value"]
+            goal_entry_object.save()
+
+            logger.info(f"Saved goal entry for {goal_name} on {yesterday}")
+
+            for property_name in goal.value:
+                goal_property_stats = get_goal_event_stats(
+                    settings.PLAUSIBLE_HOST,
+                    settings.PLAUSIBLE_SITE_ID,
+                    yesterday,
+                    goal_name,
+                    property_name,
+                )
+                for property_statistic in goal_property_stats["results"]:
+                    goal_property_object, _ = GoalProperty.objects.get_or_create(
+                        goal=goal_object,
+                        name=property_name,
+                    )
+                    (
+                        goal_property_event_object,
+                        _,
+                    ) = GoalPropertyEntry.objects.get_or_create(
```

Comment:
```suggestion
                    goal_property_event_object, _ = GoalPropertyEntry.objects.get_or_create(
```

Reply:
The black formatter would automatically format it to what it is right now. I cannot commit this suggestion 😅

Reply:
Enclose the queryset in perens that should fix it

Reply:
Still unable to format it as per the suggestion, messaged you personally to resolve this.